src/pages/Landing/index.tsx

```diff
@@ -44,8 +44,28 @@ export default () => {
     // TODO: Stretch - Use the options object to handle errors.
   );
 
+  const newUserMutation = useMutation(postNewCustomer, {
+    onMutate: async newCustomer => {
+      await queryClient.cancelQueries('customers');
+      const previousCustomers = queryClient.getQueryData('customers');
+      queryClient.setQueryData('customers', old => [...old as Customer[], newCustomer]);
+      return { previousCustomers };
+    },
+    onError: (err, newCustomer, context) => {
+      queryClient.setQueryData('todos', context?.previousCustomers)
+    },
+    onSettled: () => queryClient.invalidateQueries({ queryKey: ['customers'] })
```

Comment:
To test this functionality (the error rollback) you should go to apiUtilities.ts and set the chanceOfSuccess to 0.5 or something.

You'll find you have a bug here that duplicates values.